addons/io/org.openhab.io.homekit/src/main/java/org/openhab/io/homekit/internal/accessories/HomekitSecuritySystemImpl.java

```diff
@@ -0,0 +1,268 @@
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.io.homekit.internal.accessories;
+
+import java.util.concurrent.CompletableFuture;
+
+import org.eclipse.smarthome.core.items.GenericItem;
+import org.eclipse.smarthome.core.items.GroupItem;
+import org.eclipse.smarthome.core.items.Item;
+import org.eclipse.smarthome.core.items.ItemRegistry;
+import org.eclipse.smarthome.core.library.items.StringItem;
+import org.eclipse.smarthome.core.library.types.StringType;
+import org.eclipse.smarthome.core.types.State;
+import org.openhab.io.homekit.internal.HomekitAccessoryUpdater;
+import org.openhab.io.homekit.internal.HomekitTaggedItem;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.beowulfe.hap.HomekitCharacteristicChangeCallback;
+import com.beowulfe.hap.accessories.SecuritySystem;
+import com.beowulfe.hap.accessories.properties.CurrentSecuritySystemState;
+import com.beowulfe.hap.accessories.properties.SecuritySystemAlarmType;
+import com.beowulfe.hap.accessories.properties.TargetSecuritySystemState;
+
+/**
+ * Implements a Security System as a GroupedAccessory made up of multiple items:
+ * <ul>
+ * <li>Current Security System State: String type</li>
+ * <li>Target Security System State: String type</li>
+ * <li>Alarm Type: Switch type</li>
+ * </ul>
+ *
+ * @author Justin Mutter
+ */
+class HomekitSecuritySystemImpl extends AbstractHomekitAccessoryImpl<GroupItem>
+        implements SecuritySystem, GroupedAccessory {
+
+    private final String groupName;
+    private String currentSecuritySystemStateItemName;
+    private String targetSecuritySystemStateItemName;
+    private String alarmTypeItemName;
+
+    private Logger logger = LoggerFactory.getLogger(HomekitSecuritySystemImpl.class);
+
+    public HomekitSecuritySystemImpl(HomekitTaggedItem taggedItem, ItemRegistry itemRegistry,
+            HomekitAccessoryUpdater updater) {
+        super(taggedItem, itemRegistry, updater, GroupItem.class);
+        this.groupName = taggedItem.getItem().getName();
+    }
+
+    @Override
+    public String getGroupName() {
+        return groupName;
+    }
+
+    @Override
+    public void addCharacteristic(HomekitTaggedItem item) {
+        switch (item.getCharacteristicType()) {
+            case SECURITY_SYSTEM_CURRENT_STATE:
+                currentSecuritySystemStateItemName = item.getItem().getName();
+                break;
+
+            case SECURITY_SYSTEM_TARGET_STATE:
+                targetSecuritySystemStateItemName = item.getItem().getName();
+                break;
+
+            case SECURITY_SYSTEM_ALARM_TYPE:
+                alarmTypeItemName = item.getItem().getName();
+                break;
+
+            default:
+                logger.error("Unrecognized security system characteristic: {}", item.getCharacteristicType().name());
```

Comment:
A lot of logging is done on `error` in this PR. I think `warn` is more appropriate. See also the [Logging Guidelines](https://www.openhab.org/docs/developer/development/guidelines.html#e-logging):

* `warn` logging should only be used to inform the user that something seems to be wrong in his overall setup, but the system can nonetheless function as normal, while possibly ignoring some faulty configuration/situation. It can also be used in situations, where a code section is reached, which is not expected by the implementation under normal circumstances (while being able to automatically recover from it).
* `error` logging should only be used to inform the user that something is tremendously wrong in his setup, the system cannot function normally anymore, and there is a need for immediate action. It should also be used if some code fails irrecoverably and the user should report it as a severe bug.